src/timer.c

```diff
@@ -42,6 +46,9 @@ inline double hc_timer_get (hc_timer_t a)
 
   hc_timer_set (&hr_tmp);
 
+  #if defined(__APPLE__) && defined(MISSING_CLOCK_GETTIME)
+  return (double) (((hr_tmp.tv_sec - (a).tv_sec) * 1000) + ((double) (hr_tmp.tv_usec - (a).tv_usec) / 1000));
```

Comment:
hmm, could/should this be improved with something like this:
```
   hc_timer_t s;
 
   s.tv_sec   = hr_tmp.tv_sec   - a.tv_sec;
   s.tv_usec = hr_tmp.tv_usec - a.tv_usec;
 
   if (s.tv_usec < 0)
   {
     s.tv_sec   -= 1;
     s.tv_usec += 1000;
   }
 
   double r = ((double) s.tv_sec * 1000) + ((double) s.tv_usec / 1000);

   return r;
```
???
( I didn't test it, but if I look at the other branch - the #else - it seems to be more consistent)
we could also perform the return after the branch (because it is common code to both branches) and the variable definition of hc_timer_t s before the branches (but this isn't too important, it just would emphasize where the main differeneces between the 2 cases are, i.e. the nsec vs usec etc)

Reply:
I've merged the original version from @matrix but it would be nice if you can send in a PR with the easier to read version you've posted in the comment, too.